client/src/lib/components/SearchableSelect/SearchableSelect.svelte

```diff
@@ -1,62 +1,63 @@
 <script lang="ts">
   import type { SelectOption } from "src/definitions/form";
-  import { createEventDispatcher } from "svelte";
   import { clickOutside } from "src/lib/actions/clickOutside";
-
+  import { slugify } from "src/lib/util/format";
+  import { createEventDispatcher } from "svelte";
   export let buttonPlaceholder: string;
   export let inputPlaceholder: string;
+  export let label: string;
   export let options: Array<SelectOption>;
-
   let macthedOptions: Array<SelectOption> = options;
   let isOverlayOpen = false;
   let buttonText = buttonPlaceholder;
-
+  let selectedOption: SelectOption;
   let searchTerm: string;
-
   $: macthedOptions = options.filter((item) =>
     item.label.match(new RegExp(searchTerm, "i"))
   );
-
+  $: slug = slugify(label);
   const openOverlay = () => {
     isOverlayOpen = true;
   };
-
   const closeOverlay = () => {
     macthedOptions = options;
     isOverlayOpen = false;
   };
-
   const handleOverlayOpening = () => {
     if (!isOverlayOpen) {
       openOverlay();
       return;
     }
-
     if (isOverlayOpen) {
       closeOverlay();
       return;
     }
   };
-
   const dispatch = createEventDispatcher<{
     clickItem: SelectOption | null;
   }>();
-
   const handleClickListItem = (option: SelectOption) => {
     buttonText = option.label;
     dispatch("clickItem", option);
     closeOverlay();
   };
-
   const handleResetFilter = () => {
     macthedOptions = options;
     buttonText = buttonPlaceholder;
     dispatch("clickItem", null);
+    closeOverlay();
   };
 </script>
 
-<div class="container" use:clickOutside={{ callback: handleOverlayOpening }}>
-  <button class="fr-select" on:click={handleOverlayOpening}>{buttonText}</button
+<div use:clickOutside={{ callback: () => closeOverlay() }} class="container">
+  <label data-testId={`${slug}-label`} id={`${slug}-label`} for={slug}
```

Comment:
N.B. - Sauf erreur de ma part on ne se réfère jamais à ces test IDs avec slug dans les tests. Il semble donc qu'on puisse retirer tout ce qui a trait au slug ?

Reply:
je vais en avoir besoin pour les tests e2e mais si tu veux oui je peux les retirés de cette PR mais ils finiront pas revenir dans l'autre PR

Reply:
Je vais laisser le slug aussi pcq ça me permet de créer un `id` et un `for` ayant ce format "infromation-generales" au lieu de "Informations Générales" par exemple

Reply:
OK :+1: